ex5/ex5.c

```diff
@@ -16,6 +16,38 @@ char* msg3 = "hello world #3";
 int main()
 {
     // Your code here
-    
+       char inbuf[MSGSIZE];
+    int p[2];
+    if (pipe(p) < 0)
+    {
+        fprintf(stderr, "pipe failed\n");
+        exit(1);
+    }
+
+    int rc = fork();
+
+    if (rc < 0)
+    {
+        printf("nope\n");
+    }
+    else if (rc == 0)
+    {
+        write(p[1], msg1, MSGSIZE);
+        write(p[1], msg2, MSGSIZE);
+        write(p[1], msg3, MSGSIZE);
+        printf("inside child\n");
+    }
+    else if (rc > 0)
+    {
+        wait(NULL);
```

Comment:
The compiler warns of an implicit declaration of `wait` on line 42, check my earlier comments for more details on the topic. The program output is as expected though, so nothing major to worry about.